src/main/scala/schemas.scala

```diff
@@ -5,6 +5,13 @@ package schema
 object ScalaSchema {
   type Prim[A] = ???
 }
+
 object JsonSchema {
-  type Prim[A] = ???
+  type Prim[A] = JsonPrim[A]
+
+  sealed trait JsonPrim[A]
+  final case class JsonString(value: String) extends JsonPrim[String]
```

Comment:
Remember that we are talking about "types" here (more precisely, the atomic building blocks of a *schema*), so we dont care about values at that point. I think these `Json*` can be case objects.

Reply:
All right, `A` got me confused there. I’ll change it tomorrow morning. Thanks!